src/main/scala/is/hail/expr/ir/ExtractAggregators.scala

```diff
@@ -77,6 +77,9 @@ object ExtractAggregators {
         Code(codeConstructorArgs.map(_.setup): _*),
         AggOp.get(aggSig)
           .stagedNew(codeConstructorArgs.map(_.v).toArray, codeConstructorArgs.map(_.m).toArray)))
-      Region.scoped(constfb.result()()(_))
+
+      val rvagg = Region.scoped(constfb.result()()(_))
```

Comment:
bleh, I don't have a great answer. I think I vaguely prefer a second case match that checks for that particular aggregator and passes the type in explicitly. You can pass it through the constfb's argument and reference it as `constfb.getArg[TStruct](1)`.

Reply:
This was one thing I was considering, but went with the easier solution. I'll revisit this.